nodestream_akamai/edns/edns.py

```diff
@@ -1,57 +1,65 @@
+import asyncio
 import logging
-from ipaddress import IPv4Address, ip_address
 
 from nodestream.pipeline.extractors import Extractor
 
+from ..akamai_utils import addresses
 from ..akamai_utils.edns_client import AkamaiEdnsClient
 
+SUPPORTED_RECORD_TYPES = [
+    "A",
+    "AAAA",
+    "CNAME",
+    "NS",
+    "CAA",
+]
+
 
 class AkamaiEdnsExtractor(Extractor):
     def __init__(self, **akamai_client_kwargs) -> None:
         self.client = AkamaiEdnsClient(**akamai_client_kwargs)
         self.logger = logging.getLogger(self.__class__.__name__)
-        self.SUPPORTED_RECORD_TYPES = ["A", "AAAA", "CNAME", "NS", "CAA"]
-        self.ADDRESS_FORMAT_TO_NODE_TYPE = {
-            "ipv4": "Cidripv4",
-            "ipv6": "Cidripv6",
-            "endpoint": "Endpoint",
-        }
-
-    def _get_address_format(self, address):
+
+    def _extract_recordset(self, recordset, zone):
```

Comment:
Trying to diagnose exactly what I was seeing, I chopped this up a little to give myself more ways to play with what was going on a bit more directly.